packages/client/src/perma-cache.js

```diff
@@ -0,0 +1,396 @@
+/**
+ * A client library for the https://api.nftstorage.link/ perma-cache service. It provides a convenient
+ * interface for working with the [Raw HTTP API](https://nftstorage.link/#api-docs)
+ * from a web browser or [Node.js](https://nodejs.org/) and comes bundled with
+ * TS for out-of-the box type inference and better IntelliSense.
+ *
+ * @example
+ * ```js
+ * import { PermaCache } from 'nftstorage.link'
+ *
+ * const urls = [
+ *  'https://bafkreidyeivj7adnnac6ljvzj2e3rd5xdw3revw4da7mx2ckrstapoupoq.ipfs.nftstorage.link'
+ * ]
+ *
+ * const client = new PermaCache({ token: 'YOUR_NFT_STORAGE_TOKEN' })
+ * const permaCacheEntries = await client.put(urls)
+ * await client.delete(urls)
+ * ```
+ * @module
+ */
+
+import throttledQueue from 'throttled-queue'
+import pSettle from 'p-settle'
+import pRetry, { AbortError } from 'p-retry'
+
+import { fetch } from './platform.js'
+
+const MAX_RETRIES = 5
+// These match what is enforced server-side
+const RATE_LIMIT_REQUESTS = 100
+const RATE_LIMIT_PERIOD = 60 * 1000
+const ABORTABLE_ERRORS_STATUS = [
```

Comment:
```suggestion
// HTTP response statuses that should not be retried if encountered.
const FATAL_RESPONSE_STATUSES = [
```

Reply:
Thinking about it, isn't it just _all_ client errors (4xx)?

Reply:
I will check if in range of 4xx